### ISSUE TITLE
KeyError: 'HTTP_REFERER' when behind reverse proxy

### ISSUE THREAD

**Issue by sephrat** [morpheus65535/bazarr#455]
Bazarr is behind Nginx. I get nasty errors when doing stuff through the UI, the most explicit one being in the series editor:

```
Traceback (most recent call last):
  File "/opt/bazarr/bazarr/../libs/bottle.py", line 862, in _handle
    return route.call(**args)
  File "/opt/bazarr/bazarr/../libs/bottle.py", line 1740, in wrapper
    rv = callback(*a, **ka)
  File "/opt/bazarr/bazarr/main.py", line 137, in wrapper
    return func(*a, **ka)
  File "/opt/bazarr/bazarr/main.py", line 668, in edit_serieseditor
    ref = request.environ['HTTP_REFERER']
KeyError: 'HTTP_REFERER'
```
Another (less explicit) error is when I manual search a subtitle I get this error:
`DataTables warning: table id=search_result - Ajax error. For more information about this error, please see http://datatables.net/tn/7
`

I've narrowed down the root cause to this HTTP header:
```
add_header 'Referrer-Policy' 'no-referrer';
```
When I remove this line it works fine but I get security warnings if I audit my server (https://securityheaders.com/)

> Referrer Policy is a new header that allows a site to control how much information the browser includes with navigations away from a document and should be set by all sites.

To be honest I'm not an expert on this, I just followed tutorials but I guess removing this header is not a proper fix.

I haven't bumped into similar issues with Sonarr/Radarr so far with the same configuration in Nginx.

**Comment by morpheus65535** [comment]
You need to let your reverse proxy forward the HTTP_REFERER so Bazarr is able to send you back to your previous page once it's done processing some requests. Behind the UI, nothing is shared so they are doing everything using AJAX which doesn't bring that requirement.

Using HTTP_REFERER isn't a "bad" thing, Bazarr doesn't care where you where before on the internet, it just need to know where you were on Bazarr.

Regarding the datatables error, is it fix when you let HTTP_REFERER go trough your reverse proxy?

**Comment by morpheus65535** [comment]
Duplicate of #357

**Comment by sephrat** [comment]
OK I've changed my setting to Strict-origin-when-cross-origin and it did the trick.
The database error does indeed get fixed as well by doing so. I'm guessing the manual search somehow uses the referrer with a different error handling and the unexpected result somehow finds its way to the database query which triggers this other unexpected error.

Thanks for your quick and detailed answer!

**Comment by morpheus65535** [comment]
Thanks for sharing with us.

In fact we also get the HTTP_REFERER when doing a manual search but I should get rid of it as we don't use it. Thanks for pointing out that one! :-)

**Comment by morpheus65535** [comment]
Fixed in 9ef0ce689d058c572875746c89d50fd178260d2d
